common/lib/xmodule/xmodule/discussion_module.py

```diff
@@ -53,9 +68,26 @@ class DiscussionModule(DiscussionFields, XModule):
     js_module_name = "InlineDiscussion"
 
     def get_html(self):
+        course = self.get_course()
+        user = None
+        user_service = self.runtime.service(self, 'user')
+        if user_service:
+            user = user_service._django_user  # pylint: disable=protected-access
+        if user:
+            course_key = course.id  # pylint: disable=no-member
+            can_create_comment = has_permission(user, "create_comment", course_key)
+            can_create_subcomment = has_permission(user, "create_sub_comment", course_key)
+            can_create_thread = has_permission(user, "create_thread", course_key)
+        else:
+            can_create_comment = False
+            can_create_subcomment = False
+            can_create_thread = False
         context = {
             'discussion_id': self.discussion_id,
-            'course': self.get_course(),
+            'course': course,
+            'can_create_comment': json.dumps(can_create_comment),
+            'can_create_subcomment': json.dumps(can_create_subcomment),
+            'can_create_thread': can_create_thread,
```

Comment:
@singingwolfboy is there a reason why `json.dumps()` isn't used here like the others?